tests/test_filters.py

```diff
@@ -689,3 +740,34 @@ def test_flate_decode__not_rectangular(caplog):
     expected = get_data_from_url(url, name=name)
     assert actual_image.getvalue() == expected
     assert caplog.messages == ["Image data is not rectangular. Adding padding."]
+
+
+def test_main_decode_brotli_installed():
+    """Test the main decode function with Brotli filter using a real PDF."""
+    if importlib.util.find_spec("brotli") is None:
+        pytest.skip("brotli library not installed")
+
+    pdf_path = RESOURCE_ROOT / "brotli-test-pdfs" / "minimal-brotli-compressed.pdf"
+
+    reader = PdfReader(pdf_path)
+    page = reader.pages[0]
+    extracted_text = page.extract_text()
+
+    assert extracted_text.strip() == "Hello, Brotli!"
+
+def test_brotli_import_error_with_patch():
```

Comment:
I still do not really understand why we need this complexity. Could you please elaborate?